comp/otelcol/status/impl/status.go

```diff
@@ -0,0 +1,227 @@
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2025-present Datadog, Inc.
+
+// Package statusimpl implements the status component interface
+package statusimpl
+
+import (
+	"embed"
+	"encoding/json"
+	"fmt"
+	"gopkg.in/yaml.v3"
+	"io"
+	"net/http"
+
+	"github.com/DataDog/datadog-agent/comp/core/config"
+	statusComponent "github.com/DataDog/datadog-agent/comp/core/status"
+	ddflareextension "github.com/DataDog/datadog-agent/comp/otelcol/ddflareextension/def"
+	status "github.com/DataDog/datadog-agent/comp/otelcol/status/def"
+	apiutil "github.com/DataDog/datadog-agent/pkg/api/util"
+	"github.com/DataDog/datadog-agent/pkg/util/prometheus"
+)
+
+//go:embed status_templates
+var templatesFS embed.FS
+
+// Requires defines the dependencies of the status component.
+type Requires struct {
+	Config config.Component
+}
+
+// Provides contains components provided by status constructor.
+type Provides struct {
+	Comp           status.Component
+	StatusProvider statusComponent.InformationProvider
+}
+
+type statusProvider struct {
+	Config         config.Component
+	client         *http.Client
+	receiverStatus map[string]interface{}
+	exporterStatus map[string]interface{}
+}
+
+type prometheusRuntimeConfig struct {
+	Service struct {
+		Telemetry struct {
+			Metrics struct {
+				Readers []struct {
+					Pull struct {
+						Exporter struct {
+							Prometheus struct {
+								Host string
+								Port int
+							}
+						}
+					}
+				}
+			}
+		}
+	}
+}
+
+// NewComponent creates a new status component.
+func NewComponent(reqs Requires) Provides {
+	comp := statusProvider{
+		Config: reqs.Config,
+		client: apiutil.GetClient(false),
+		receiverStatus: map[string]interface{}{
+			"spans":           0.0,
+			"metrics":         0.0,
+			"logs":            0.0,
+			"refused_spans":   0.0,
+			"refused_metrics": 0.0,
+			"refused_logs":    0.0,
+		},
+		exporterStatus: map[string]interface{}{
+			"spans":          0.0,
+			"metrics":        0.0,
+			"logs":           0.0,
+			"failed_spans":   0.0,
+			"failed_metrics": 0.0,
+			"failed_logs":    0.0,
+		},
+	}
+	return Provides{
+		Comp:           comp,
+		StatusProvider: statusComponent.NewInformationProvider(comp),
+	}
+}
+
+// Name returns the name
+func (s statusProvider) Name() string {
+	return "OTel Agent"
+}
+
+// Section return the section
+func (s statusProvider) Section() string {
+	return "OTel Agent"
+}
+
+func (s statusProvider) getStatusInfo() map[string]interface{} {
+	statusInfo := make(map[string]interface{})
+
+	values := s.populateStatus()
+
+	statusInfo["otelAgent"] = values
+
+	return statusInfo
+}
+
+func getPrometheusURL(extensionResp ddflareextension.Response) (string, error) {
+	var runtimeConfig prometheusRuntimeConfig
+	if err := yaml.Unmarshal([]byte(extensionResp.RuntimeConfig), &runtimeConfig); err != nil {
+		return "", err
+	}
```

Comment:
Out for my curiosity, could we avoid this second yaml parse of the HTTP body? I imagine that we are returning string representation of the extension config, right?

Reply:
We're returning string representation of the extension config but it's pretty difficult to parse in string form due to formatting and the non-unique key names